pkg/apiroutes/templates.go

```diff
@@ -355,3 +357,22 @@ func BatchPatchTemplateRepos(conID string, operations []RepoOperation) ([]SubRes
 
 	return subResponsesFromBatchOperation, nil
 }
+
+// httpRequestWithRetryOnLock : Retries a request until either the maxHTTPRetries limit has been hit or the response status code is not equal to 423 (templates locked HTTP code)
+func httpRequestWithRetryOnLock(httpClient utils.HTTPClient, originalRequest *http.Request, connection *connections.Connection) (*http.Response, *sechttp.HTTPSecError) {
```

Comment:
if you did want to look at tests for this, an interface with a Calls counter could come in use? See halfway down this page https://quii.gitbook.io/learn-go-with-tests/go-fundamentals/mocking. Could be overkill, but then could come in useful in other areas too.

Reply:
You could then modify the response on dispatchRequest based on the number of calls

Reply:
This is a good suggestion but I think at the moment this is overkill. I have added tests which ensure that the correct response and errors are passed back as this is essentially just a secHTTP wrapper.